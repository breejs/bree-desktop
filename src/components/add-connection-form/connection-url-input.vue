<script setup>
import { ref, computed } from 'vue';

import useDebounceInput from '@/composables/debounce-input';

const props = defineProps({
  value: {
    type: String,
    default: 'http://localhost:62893'
  }
});

const emit = defineEmits(['update:value']);

const input = ref(null);

const { inputValue, valid, classes } = useDebounceInput(
  props.value,
  'http://localhost:62893',
  input,
  emit,
  (value) =>
    /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z\d.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z\d.-]+)((?:\/[+~%/.\w-_]*)?\??[-+=&;%@.\w_]*#?[.!/\\\w]*)?)/.test(
      value
    )
);

defineExpose({
  valid
});
</script>

<template lang="pug">
.form-floating
  input#connection-url-input.form-control(
    name='url',
    type='url',
    placeholder='http://localhost:62893',
    required,
    v-model='inputValue',
    :class='classes',
    ref='input'
  )
  label(for='connection-url-input')
    | URL*
  .invalid-feedback
    | Enter a valid URL
</template>
