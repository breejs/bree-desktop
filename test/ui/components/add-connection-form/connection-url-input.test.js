import { storyTests } from '../../_story-tests';

import * as stories from '@/components/add-connection-form/connection-url-input.stories';

storyTests(stories);
