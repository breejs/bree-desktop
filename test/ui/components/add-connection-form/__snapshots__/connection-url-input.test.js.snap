// Vitest Snapshot v1

exports[`story snapshots > Default 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-url-input\\" name=\\"url\\" type=\\"url\\" placeholder=\\"http://localhost:62893\\" required=\\"\\"><label for=\\"connection-url-input\\">URL*</label>
    <div class=\\"invalid-feedback\\">Enter a valid URL</div>
  </div>
</form>"
`;

exports[`story snapshots > Invalid 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-url-input\\" name=\\"url\\" type=\\"url\\" placeholder=\\"http://localhost:62893\\" required=\\"\\"><label for=\\"connection-url-input\\">URL*</label>
    <div class=\\"invalid-feedback\\">Enter a valid URL</div>
  </div>
</form>"
`;

exports[`story snapshots > Valid 1`] = `
"<form novalidate=\\"\\">
  <div class=\\"form-floating\\"><input class=\\"form-control\\" id=\\"connection-url-input\\" name=\\"url\\" type=\\"url\\" placeholder=\\"http://localhost:62893\\" required=\\"\\"><label for=\\"connection-url-input\\">URL*</label>
    <div class=\\"invalid-feedback\\">Enter a valid URL</div>
  </div>
</form>"
`;
